astrobase/periodbase/tls.py

```diff
@@ -0,0 +1,380 @@
+
+        The descriptions are here:
+        https://transitleastsquares.readthedocs.io/en/latest/Python%20interface.html#return-values
+
+        The remaining resultdict is:
+
+            resultdict = {
+                'tlsresult':tlsresult,
+                'bestperiod': the best period value in the periodogram,
+                'bestlspval': the periodogram peak associated with the best period,
+                'nbestpeaks': the input value of nbestpeaks,
+                'nbestlspvals': nbestpeaks-size list of best period peak values,
+                'nbestperiods': nbestpeaks-size list of best periods,
+                'lspvals': the full array of periodogram powers,
+                'periods': the full array of periods considered,
+                'tlsresult': Astropy tls result object (BoxLeastSquaresResult),
+                'tlsmodel': Astropy tls BoxLeastSquares object used for work,
+                'method':'tls' -> the name of the period-finder method,
+                'kwargs':{ dict of all of the input kwargs for record-keeping}
+            }
+    """
+
+    if not magsarefluxes:
+        errmsg = ('IDK if transitleastsquares supports magnitudes. '
+                  'many of its utilities would probably break.')
+        raise NotImplementedError(errmsg)
+
+    if nworkers is None:
+        nworkers = NCPUS
+
+    # get rid of nans first and sigclip
+    stimes, smags, serrs = sigclip_magseries(times, mags, errs,
+                                             magsarefluxes=magsarefluxes,
+                                             sigclip=sigclip)
+
+    # make sure there are enough points to calculate a spectrum
+    if not (len(stimes) > 9 and len(smags) > 9 and len(serrs) > 9):
+        LOGERROR('no good detections for these times and mags, skipping...')
+        resultdict = {
+            'tlsresult':npnan,
+            'bestperiod':npnan,
+            'bestlspval':npnan,
+            'nbestpeaks':nbestpeaks,
+            'nbestinds':None,
+            'nbestlspvals':None,
+            'nbestperiods':None,
+            'lspvals':None,
+            'periods':None,
+            'method':'tls',
+            'kwargs':{'startp':startp,
+                      'endp':endp,
+                      'tlsoversample':tlsoversample,
+                      'tlsntransits':tlsmintransits,
+                      'periodepsilon':periodepsilon,
+                      'nbestpeaks':nbestpeaks,
+                      'sigclip':sigclip,
+                      'magsarefluxes':magsarefluxes}
+        }
+        return resultdict
+
+    if endp is None:
+        # out to 99% of the baseline. (for two transits).
+        endp = 0.99*(np.nanmax(stimes) - np.nanmin(stimes))
+
+    if isinstance(errs,np.ndarray):
+        model = transitleastsquares(stimes, smags, serrs)
+    else:
+        model = transitleastsquares(stimes, smags)
+
+    # run periodogram
+    tlsresult = model.power(use_threads=nworkers, show_progress_bar=False,
```

Comment:
Add all of these kwargs to the `tls_parallel_pfind` function kwargs so they're configurable. Maybe prefix them with `tls_`